Sources/ApolloTestSupport/MockGraphQLServer.swift

```diff
@@ -0,0 +1,102 @@
+@testable import Apollo
+import XCTest
+
+/// A `MockGraphQLServer` can be used during tests to check whether expected GraphQL requests are received, and to respond with appropriate test data for a particular request.
+///
+/// You usually create a  mock server in the test's `setUpWithError`, and use it to initialize a `MockNetworkTransport` that is in turn used  to initialize an `ApolloClient`:
+///  ```
+/// let server = MockGraphQLServer()
+/// let networkTransport = MockNetworkTransport(server: server, store: store)
+/// let client = ApolloClient(networkTransport: networkTransport, store: store)
+///  ```
+/// A mock server should be configured to expect particular operation types, and invokes the passed in request handler when a request of that type comes in. Because the request allows access to `operation`, you can return different responses based on query variables for example:
+
+/// ```
+/// let serverExpectation = server.expect(HeroNameQuery.self) { request in
+///   [
+///     "data": [
+///       "hero": [
+///         "name": request.operation.episode == .empire ? "Luke Skywalker" : "R2-D2",
+///         "__typename": "Droid"
+///       ]
+///     ]
+///   ]
+/// }
+/// ```
+/// By default, expectations returned from `MockGraphQLServer` only expect to be called once, which is similar to how other built-in expectations work. Unexpected fulfillments will result in test failures. But if multiple fulfillments  are expected, you can use the standard `expectedFulfillmentCount` property to change that. For example, some of the concurrent tests expect the server to receive the same number of request as the number of invoked fetch operations, so in that case we can use:
+
+/// ```
+/// serverExpectation.expectedFulfillmentCount = numberOfFetches
+/// ```
+public class MockGraphQLServer {
```

Comment:
My thinking was that `MockNetworkTransport` is an artifact of the current split between `NetworkTransport` and the request chain, and once we switch over to a request chain model completely we won't have `NetworkTransport` at all.

Even today, having a fixed `TestInterceptorProvider` may be too limiting, so I can imagine wanting to use `MockGraphQLServerInterceptor` directly to set up a chain for specific testing scenarios.

Reply:
True. One other place it might make sense to do this is as a subclass of `URLSessionClient` - that's almost certainly going to stick around no matter what, and then it sends things through the real interceptor chain as well.